content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md

```diff
@@ -751,26 +751,154 @@ with the response:
+    - A schema with no type and x-kubernetes-preserve-unknown-fields set to true
+    - An array where the items schema is of an "unknown type"
+    - An object where the additionalProperties schema is of an "unknown type"
+	
+
+Only property names of the form `[a-zA-Z_.-/][a-zA-Z0-9_.-/]*` are accessible.
+Accessible property names are escaped according to the following rules when accessed in the expression:
+
+| property name          | escaped to             |
+| -----------------------| -----------------------|
+| `__`                   | `__underscores__`      |
+| `.`                    | `__dot__`              |
+| `-`                    | `__dash__`             |
+| `/`                    | `__slash__`            |
+| [CEL RESERVED keyword](https://github.com/google/cel-spec/blob/v0.6.0/doc/langdef.md#syntax)   | `__{keyword}__`        |
+
+  Examples:
+
+    - Rule accessing a property named "namespace": {"rule": "self.__namespace__ > 0"}
+    - Rule accessing a property named "x-prop": {"rule": "self.x__dash__prop > 0"}
+    - Rule accessing a property named "redact__d": {"rule": "self.redact__underscores__d > 0"}
+    
+  
+Equality on arrays with `x-kubernetes-list-type` of `set` or `map` ignores element order, i.e. [1, 2] == [2, 1].
+Concatenation on arrays with x-kubernetes-list-type use the semantics of the list type:
+ - `set`: `X + Y` performs a union where the array positions of all elements in `X` are preserved and
+      non-intersecting elements in `Y` are appended, retaining their partial order.
+ - `map`: `X + Y` performs a merge where the array positions of all keys in `X` are preserved but the values
+   are overwritten by values in `Y` when the key sets of `X` and `Y` intersect. Elements in `Y` with
+   non-intersecting keys are appended, retaining their partial order.
+
+
+Rules may be written at the root of an object, and may make field selection into any fields
+declared in the OpenAPIv3 schema of the CRD as well as `apiVersion`, `kind`, `metadata.name` and
+`metadata.generateName`. This includes selection of fields in both the `spec` and `status` in the
+same expression, e.g. `self.status.quantity <= self.spec.maxQuantity`. Because CRDs only allow the
+`name` and `generateName` to be declared in the `metadata` of an object, these are the only
+metadata fields that may be validated using CEL validator rules. For example,
+`self.metadata.name.endsWith('mySuffix')` is allowed, but `size(self.metadata.labels) < 3` it not
+allowed. The limit on which `metadata` fields may be validated is an intentional design choice
+(that aims to allow for generic access to labels and annotations across all kinds) and applies to
+all validation mechanisms (e.g. the OpenAPIV3 `maxItems` restriction), not just CEL validator
+rules. xref rule 4 in [specifying a structural schema](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#specifying-a-structural-schema).
```

Comment:
I'd prefer to omit this paragraph. I the below "additionally" section we already mention the metadata access limit. It rather expand that (but only slightly) and not have this long explanation.

Reply:
I have rewrite few sentences about it and moved upper